### PR TITLE
Fix ldap check

### PR DESCRIPTION
**Pull request by a-breid-ericsson**
### Applicable Issues
Front end not working in EI (Internet Explorer), sometimes displays subscription owner names even when ldap is disabled.

### Description of the Change
Solved problem with IE.
ldapEnabled is defined in one place and may be accessed using getter and setters.
Minor improvements and small fixes in code.

### Benefits
It works.

### Possible Drawbacks
People may still use IE

### Sign-off

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

Signed-off-by:

### CODE REVIEW

File: src/main/resources/static/js/login.js [hunk 1]
```diff
@@ -1,28 +1,13 @@
 jQuery(document).ready(function () {
+    function reroutToSubscritpion() {
+        checkBackendSecured();
+        // if (ldapEnabled == false || (ldapEnabled == true && currentUser != null)) {
+        if (isLdapEnabled() && currentUser != null && currentUser != undefined || !isLdapEnabled()) {
```
Comment (tobiasake) [inline src/main/resources/static/js/login.js]: Why is last "!isLdapEnabled()" needed? It should be first in the if statement,, no need of doing the other checks before if isLdapEnabled is disabled.

File: src/main/resources/static/js/subscription.js [hunk 2]
```diff
@@ -467,81 +458,74 @@ jQuery(document).ready(function () {
                     "render": function (data, type, row, meta) {
                         if (data == undefined || row == undefined) {
                             window.logMessages("Error: Subscription data is not defined");
-                            return ''
+                            return '';
                         }
-                        subscriptionOwner = row.ldapUserName
-                        subscriptionName = data.subscriptionName
+                        subscriptionOwner = row.ldapUserName;
```
Comment (tobiasake) [inline src/main/resources/static/js/subscription.js]: What if ldapUsername don't exist in Subscription?

Comment (a-breid-ericsson) [reply src/main/resources/static/js/subscription.js]: subscriptionOwner will be undefined, there is a check for that when determing if owner exist for subscription.

File: src/main/resources/static/js/login.js (hunk 1 above)
Comment (a-breid-ericsson) [reply src/main/resources/static/js/login.js]: Done. It should only be !isLdapEnabled() cause if ldap is disabled then we should not show login page, and if ldap is not disabled then it must of course be true, så the other check for ldap is not needed.

File: src/main/resources/static/js/login.js [hunk 3]
```diff
@@ -1,28 +1,12 @@
 jQuery(document).ready(function () {
+    function reroutToSubscritpion() {
```
Comment (Christoffer-Cortes) [inline src/main/resources/static/js/login.js]: rerouteToSubscription

File: src/main/resources/static/js/login.js (hunk 1 above)
Comment (evasiba-ericsson) [inline src/main/resources/static/js/login.js]: Please group the logical expression used here. I assume that isLdapEnabled() && currentUser != null && currentUser != undefined  is one group and  !isLdapEnabled() is the second group. If that is the case add parathesis around the second group or make it a variable.

The expression is confusing at the first glance and it is frustrating to require to much attention to understand it.

Comment (a-breid-ericsson) [reply src/main/resources/static/js/login.js]: Done, reused code already written for other function.

File: src/main/resources/static/js/login.js (hunk 3 above)
Comment (a-breid-ericsson) [reply src/main/resources/static/js/login.js]: Done

File: src/main/resources/static/js/common.js [hunk 4]
```diff
@@ -2,19 +2,40 @@ var router = new Navigo(null, true, '#');
...
         return url;
     }
-    var delimiter = "";
-    var parameterKey = "backendname";
 
-    if (url.includes("?")) {
+    // If String includes function does not exist (IE) add function.
+    if (!String.prototype.includes) {
+        String.prototype.includes = function (str) {
+            return this.indexOf(str) !== -1;
```
[the remark this answers is not in the document]
Comment (a-breid-ericsson) [reply src/main/resources/static/js/common.js]: Done. Although I was considering naming it isSubstring, but I believe everyone knows in coding what needle in haystack means.

**Pull request merged**